pkg/cmd/cli/cmd/login/helpers.go

```diff
@@ -108,8 +110,25 @@ func getHostPort(hostURL string) (string, string, *url.URL, error) {
 	return host, port, parsedURL, err
 }
 
-func whoAmI(client *client.Client) (*api.User, error) {
+func whoAmI(clientConfig *restclient.Config) (*api.User, error) {
+	client, err := client.New(clientConfig)
+
 	me, err := client.Users().Get("~")
+
+	// if we're talking to kube (or likely talking to kube),
+	if kerrors.IsNotFound(err) {
+		switch {
+		case len(clientConfig.BearerToken) > 0:
+			// the user has already been willing to provide the token on the CLI, so they probably
+			// don't mind using it again if they switch to and from this user
+			return &api.User{ObjectMeta: kapi.ObjectMeta{Name: clientConfig.BearerToken}}, nil
```

Comment:
> this will make the token show up unredacted when using the various config commands :-/

I commented why I thought that was a reasonable thing to do.  The only way to get here was to already put the output on your CLI.